### PR TITLE
audit(memory): first Harden-era sweep - 2 lessons graduated into gates

### PR DESCRIPTION
**Pull request by gmickel**
## Summary

First Harden-era memory audit (`/flow-next:audit bug/build-errors`, interactive) - the 3.4.5 feature dogfooding itself on the repo that shipped it. 29 entries + 44 glossary terms audited; 2 lessons graduated into enforced gates, the R2 ask-block trio consolidated, 9 entries updated, 15 kept.

## What changed

- **New CI gate** `plugins/flow-next/tests/test_skill_prose_flowctl_surface.py`: every `$FLOWCTL <subcommand>` in canonical skill prose must be a registered flowctl subcommand. Includes a vacuity guard (extraction must find >20 invocations) and a negative self-test (provably fires on a bogus subcommand). Graduates `skill-prose-must-match-real-flowctl` - re-taught by fn-59, fn-68, fn-82, and fn-122 over six weeks; now demoted to a pointer at the gate.
- **Pointer-demotion**: `sync-codexsh-tool-substitution` demoted to point at the existing sync-codex.sh hard-fail guard family (duplication guard found the class already enforced; no new artifact per contract).
- **Consolidation**: the three R2 ask-block entries (same defect class, same function, same validator blind spot) merged into the 06-27 canonical; unique prevention rules absorbed as marked bullets; 2 files removed; all `related_to` references re-pointed.
- **9 updates**: drifted line refs, module fields for html-lens split-outs, detectvalidate's falsified `--epics-file` cascade section corrected to post-3.0.0 reality, version-bump entry's dead `commands/flow-next/` path fixed.
- **CLAUDE.md**: one-line `.flow/memory/` discoverability addition (Phase 6).

## Not in this PR

- `template-rewrite-env-var-cascade` mark-fresh stamp - blocked by flowctl mark-* rejecting the legacy `audit_refocused_from` field (contradicts the preserve-unknown-fields contract); follow-up task filed.
- A manifest-count consistency test (skill-adding-version-bump documents the gap but is below the Harden recurrence threshold).

## Verification

- Full suite: 130 files, 2344 tests, 0 failures (includes the new gate).
- Both `hardened_into` gate-refs grep verbatim in their artifacts (the next audit's liveness check depends on this).
- `flowctl memory list --status hardened` shows exactly the 2 demoted entries; default list/search now excludes them.

## Version note

No bump - memory-store state + one test file; staged nothing under Unreleased (no plugin behavior change).

### CODE REVIEW

File: plugins/flow-next/tests/test_skill_prose_flowctl_surface.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+SKILLS_DIR = REPO_ROOT / "plugins" / "flow-next" / "skills"
+FLOWCTL_PY = REPO_ROOT / "plugins" / "flow-next" / "scripts" / "flowctl.py"
+
+# `$FLOWCTL <token>` / `"$FLOWCTL" <token>` / `${FLOWCTL} <token>`.
+# The token must start with a lowercase letter, so flags (`--help`) never
+# match; an inline-code mention like ``` `$FLOWCTL` is the CLI ``` never
+# matches either because a backtick, not whitespace, follows the variable.
+_INVOCATION = re.compile(
+    r'(?:"\$\{?FLOWCTL\}?"|\$\{?FLOWCTL\}?)\s+([a-z][a-z0-9-]*)'
+)
```
[the remark this answers is not in the document]
Comment (gmickel) [reply plugins/flow-next/tests/test_skill_prose_flowctl_surface.py:33]: Declining (P2, measured). Extending the scan to bare `flowctl` was tested against the current tree: it produces ~90 false positives across 15+ token classes because bare `flowctl` in skill prose is overwhelmingly descriptive English ('flowctl is', 'flowctl plumbing', 'flowctl helper', 'flowctl owns') - the gate would become a noise generator on day one. The $FLOWCTL form is the executable-invocation shape agents copy and run, which is exactly the surface the graduated lesson covers; the deliberate-scope note is in the test docstring and the demoted entry remains the pointer explaining the boundary.

File: .flow/memory/bug/build-errors/sync-codexsh-tool-substitution-needs-2026-05-18.md [hunk 2]
```diff
@@ -9,7 +9,11 @@ problem_type: build-error
 symptoms: "Codex impl-review NEEDS_WORK cycles: injection inside tables/code-blocks, contradictions with auto-fix mandates, structured-tool prose surviving token rewrite, anti-patterns inverted on Codex"
 root_cause: "Token-only AskUserQuestion → plain-text-prompt rewrite leaves Claude-specific structured-tool prose (multiSelect, blocking-question, JSON questions array, deferred-tool schema-loader) intact, and 'first non-negative occurrence' injection lands inside tables, code blocks, and deterministic Ralph branches"
 resolution_type: fix
+status: hardened
+hardened_into: scripts/sync-codex.sh#Explore-dispatch transform (fn-100 R12) should have rewritten these -- hard-fail guard family for unrewritten Claude-only phrases; CLAUDE.md checklist mandates sync x2 on every canonical edit
```
[the remark this answers is not in the document]
Comment (gmickel) [reply .flow/memory/bug/build-errors/sync-codexsh-tool-substitution-needs-2026-05-18.md:13]: Declining (P2, by-design). The demotion rationale is not the single Explore-dispatch guard: the gate-ref names the hard-fail guard FAMILY in sync-codex.sh, and the broader authoring rule lives in the CLAUDE.md skill-editing checklist - which is ALWAYS loaded into every session, a strictly stronger delivery channel than memory-scout's conditional injection. The body stays on disk (searchable via --status hardened) as provenance, and if the class ever recurs the audit's gate-liveness pass proposes un-graduation via one mark-fresh command. Retiring the 14k-token context injection while the always-on checklist carries the rule is the intended trade.
